### ISSUE TITLE
Mount failed on ECS Fargate container

### ISSUE THREAD

**Issue by harikishore23** [awslabs/mountpoint-s3#450]
### Mountpoint for Amazon S3 version

mountpoint-s3 v1.0.0

### AWS Region

eu-west-2

### Describe the running environment

Running ECS Fargate container - Linux.

Dockerfile.

FROM public.ecr.aws/amazoncorretto/amazoncorretto:17-al2-jdk
RUN curl "https://s3.amazonaws.com/mountpoint-s3-release/latest/x86_64/mount-s3.rpm" -o "mount-s3.rpm"
RUN yum install -y mount-s3.rpm && yum install jq -y
RUN mkdir s3_bucket
COPY app.jar app.jar
COPY server-start.sh /
EXPOSE 8003
ENTRYPOINT ["sh", "/server-start.sh"]

server-start.sh
mount-s3 BUCKET_NAME s3_bucket

- Task has permission to access S3.
- No permissions set on S3 bucket.

### What happened?

Container started successfully but the mount failed to happen.

### Relevant log output

```shell
August 11, 2023 at 19:02 (UTC+1:00)     Error: Failed to create mount process
August 11, 2023 at 19:02 (UTC+1:00)	fuse: device not found, try 'modprobe fuse' first
August 11, 2023 at 19:02 (UTC+1:00)	Error: Failed to create FUSE session
August 11, 2023 at 19:02 (UTC+1:00)	Caused by:
```

**Comment by blankynap** [comment]
@harikishore23  are you sure ECS Fargate supports mountpoint-s3 and vice versa?

**Comment by harikishore23** [comment]
> @harikishore23 are you sure ECS Fargate supports mountpoint-s3 and vice versa?

After more digging into this repo, looks like Fargate isn't supported.

`These elevated privileges aren't available in AWS-managed container environments like [AWS Fargate](https://aws.amazon.com/fargate/). They are available in [Amazon Elastic Container Service (ECS)](https://aws.amazon.com/ecs/), but not enabled by default. `

https://github.com/awslabs/mountpoint-s3/tree/main/docker

Looks like mountpoint-s3 works only with EC2 + ECS.

**Comment by jamesbornholt** [comment]
Yeah, Mountpoint requires additional permissions that Fargate isn't able to provide. For now, ECS + EC2 is our recommended option, but we're looking into what else might be possible here. I'm going to leave this open for other folks who might try Fargate and run into the same problem.

**Comment by francisreyes-tfs** [comment]
And I guess it doesn't work in CloudShell either, same error

**Comment by rdzidziguri** [comment]
This would simplify the release process so much as, in many cases, many containers only require updating src. For containers using PHP, for example, this could be as easy as restarting the container without rebuilding it. S3 with ECS Fargate would be a huge step forward.

**Comment by pippolino** [comment]
UP

**Comment by matthiasdebernardini** [comment]
I have an webserver on fargate and I wanted to use this for it, but looks like I wont be able to. 

Should I just use aws-sdk-s3 to replicate this behavior?

**Comment by ebmeierj** [comment]
upvote

**Comment by psantus** [comment]
upvote!

**Comment by swkim86** [comment]
upvote

**Comment by Haebuk** [comment]
please

**Comment by hellosunil** [comment]
upvote!

**Comment by dannycjones** [comment]
Hey folks, please use the 👍 / thumbs up reaction on the main post as this is how we measure interest in the feature request!

**Comment by AugustDev** [comment]
Still a problem

**Comment by Menahem1** [comment]
Any news ?

**Comment by muddyfish** [comment]
We don't currently have any news to share regarding this feature request

**Comment by binChris** [comment]
Any updates?

I don't understand how a product made by AWS Labs does not support a mainstream AWS platform like ECS Fargate.

**Comment by yu-shiba** [comment]
Great news!
> Announcing Amazon S3 Files, making S3 buckets accessible as file systems  

https://aws.amazon.com/jp/about-aws/whats-new/2026/04/amazon-s3-files/

**Comment by ankit-kalyani** [comment]
Hello all, we want to know if your use cases for this ask have been addressed through the S3 Files launch and the integration with AWS Fargate. If not, it would be great to get more details on the specific use cases where you'd prefer to use Mountpoint with AWS Fargate.
